micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -218,11 +218,10 @@ void checkAndBindMetrics(MeterRegistry registry) {
                             registeredMeterIds.remove(otherId);
                         }
                         // Check if already exists
-                        else if (tags.size() == meterTagsWithCommonTags.size())
+                        else if (tags.size() == meterTagsWithCommonTags.size()) {
                             if (tags.containsAll(meterTagsWithCommonTags))
                                 return;
-                            else
-                                break;
```

Comment:
This early break would result in re-registering a meter that may already be registered, depending on the order of iteration.